File: app/src/main/java/com/pij/zworkout/ZworkoutApplication.java
package com.pij.zworkout;

import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;

/**
 * <p>Created on 01/03/2018.</p>
 *
 * @author Pierrejean
 */

public class ZworkoutApplication extends DaggerApplication {
    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        AppComponent appComponent = DaggerAppComponent.builder().application(this).build();
        appComponent.inject(this);
        return appComponent;
    }
}

File: app/src/main/java/com/pij/zworkout/ZworkoutApplication.kt
package com.pij.zworkout

import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication

/**
 *
 * Created on 01/03/2018.
 *
 * @author Pierrejean
 */

class ZworkoutApplication : DaggerApplication() {

    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        val appComponent = DaggerAppComponent.builder().application(this).build()
        appComponent.inject(this)
        return appComponent
    }
}
